### PR TITLE
Rename `logs` index mode to `logsdb`

### LINKED ISSUE #111049: Rename `logs` index mode to `logsdb`
### Description

We discussed the topic of naming regarding LogsDB and index mode `logs` and came up with the decision to stick to using just `logsdb` everywhere. The reason behind this decision is to make it easier for users and stakeholders to understand what we refer to without having multiple names for the same thing.

### PR DESCRIPTION
**Pull request by salvatore-campagna**
For consistency and to avoid multiple names for the same thing we decided to
rename the index mode to `logsdb`.

Resolves #111049

### CODE REVIEW

**Comment by salvatore-campagna** [comment]
Before we merge it we need to change the index mode in Rally tests too both stateful and non-stateful.

File: docs/reference/data-streams/logs.asciidoc [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [[logs-data-stream]]
 == Logs data stream
 
-preview::[Logs data streams and the logs index mode are in tech preview and may be changed or removed in the future. Don't use logs data streams or logs index mode in production.]
```
Comment (salvatore-campagna) [inline docs/reference/data-streams/logs.asciidoc:4]: @shainaraskas I had to change this after #110932 and #110054

**Review by lkts — APPROVED** [review]

**Review by dnhatn — APPROVED** [review]
LGTM2. Thanks @salvatore-campagna.

**Review by shainaraskas — APPROVED** [review]
docs lgtm

**Comment by lkts** [comment]
`rest-api-spec/test/logsdb/20_mapping.yml` needs to be updated too?

**Comment by ruflin** [comment]
@jsoriano For awareness but I think it does not change anything in elastic-package at the moment?

**Comment by jsoriano** [comment]
> @jsoriano For awareness but I think it does not change anything in elastic-package at the moment?

By now we only referenced this index mode here https://github.com/elastic/elastic-package/pull/1979

**Comment by salvatore-campagna** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.15|[](https://github.com/elastic/elasticsearch/pull/111098)|


### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**